lib/core/src/lib/notifications/components/notification-history.component.spec.ts

```diff
@@ -182,4 +184,119 @@ describe('Notification History Component', () => {
+        fixture.detectChanges();
+
+        expect(component.unreadNotifications).toEqual([]);
+    });
+
+    it('should set unreadNotifications to an empty array when all notifcations are read', () => {
+        const notifications = [
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Moved 1 item.'],
+                read: true
+            },
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Copied 1 item.'],
+                read: true
+            }
+        ];
+        storage.setItem(NotificationHistoryComponent.NOTIFICATION_STORAGE, JSON.stringify(notifications));
+        fixture.detectChanges();
+
+        expect(component.unreadNotifications.length).toEqual(0);
+        expect(component.unreadNotifications).toEqual([]);
+    });
+
+    it('should set unreadNotifications by filtering notifications where read is false', () => {
+        const notifications = [
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: '',
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Moved 1 item.'],
+                read: false
+            },
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Copied 1 item.'],
+                read: true
+            }
+        ];
+        storage.setItem(NotificationHistoryComponent.NOTIFICATION_STORAGE, JSON.stringify(notifications));
+        fixture.detectChanges();
+
+        expect(component.unreadNotifications.length).toEqual(1);
+        expect(component.unreadNotifications[0].read).toEqual(false);
+    });
+
+    it('should return badgeVisibility true when there are unread notifications', () => {
+        component.unreadNotifications = [
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Moved 1 item.'],
+                read: false
+            },
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Copied 1 item.'],
+                read: false
+            }
+        ];
+
+        const result = component.badgeVisibility();
+        const matIconDebugElement = fixture.debugElement.query(By.css('[matBadge]')).nativeElement;
```

Comment:
This part is repeated: fixture.debugElement.query(By.css('[matBadge]')).nativeElement;
Please move it to function and reuse.

Reply:
moved selector logic to a method